lib/src/web_ui/code_input.dart

```diff
@@ -101,6 +121,43 @@ class CodeInput {
     parenListener(missingParens);
   }
 
+  /// Determines the operation at the last open parens
+  List _currOp(String inputText, int position, [int fromLast = 1]) {
+    List<String> splitLines = inputText.substring(0, position).split("\n");
+    //The first item indicates the word that was matched, the second item indicates
+    //if that word is the full string(true) or is in progress of being written out(false)
+    bool multipleLines = false;
+    String refLine;
+    int totalMissingCount = 0;
+
+    for (refLine in splitLines.reversed) {
+      totalMissingCount += countParens(refLine) ?? 0;
+      // Find the first line with an open paren but no close paren
+      if (totalMissingCount >= fromLast) break;
+      multipleLines = true;
+    }
+    //if there are not enough open parentheses, return the default output value
+    if (totalMissingCount >= fromLast) {
+      int strIndex = refLine.indexOf("(");
+      while (strIndex != -1 && strIndex < (refLine.length - 1)) {
+        int nextClose = refLine.indexOf(")", strIndex + 1);
+        int nextOpen = refLine.indexOf("(", strIndex + 1);
+        // Find the open paren that corresponds to the missing closed paren
+        if (totalMissingCount > fromLast) {
+          totalMissingCount -= 1;
+        } else if (nextOpen == -1 || nextClose == -1 || nextOpen < nextClose) {
+          //Assuming the word is right after the parens
+          List splitRef =
+              refLine.substring(strIndex + 1).split(RegExp("[ ()]+"));
```

Comment:
Not sure what this regular expression is actually doing. I recommend including a comment describing the meaning whenever you use all but the most trivial regexes, since they are notoriously difficult to understand.